### ISSUE TITLE
Add option how to handle errors

### ISSUE THREAD

**Issue by thebrownfox** [drewloomer/hook-shell-script-webpack-plugin#18]
Currently there's only watch that's telling the plugin not to stop the execution. In some cases the line bellow stops execution because of harmless terminal output, like `Error: [HookShellScriptPlugin] Error while running `node initPlugins.mjs`: Debugger attached.`.

```
 this._procs[key].stderr.on('data',...)
```

Can I send quick PR?

**Comment by drewloomer** [comment]
Yeah send a PR!

**Comment by rrthomas** [comment]
I too would like to see this PR! In the mean time, I'm not a webpack expert, but the simplest solution seems to me simply to remove line 82 of `index.js` that raises an error on stderr output, and instead rely on the exit status of the command to indicate an error. Would that change be OK?

**Comment by mykeels** [comment]
```ts
const shellScriptPlugin = new HookShellScriptPlugin({
    beforeCompile: ['foo'],
  });
shellScriptPlugin._onScriptError = () => {};
```

worked for ignoring errors.

In my case, it was a console.warn that it saw as an error, so ignoring it helped.

**Comment by rrthomas** [comment]
Thanks @mykeels for the workaround.

However, I noticed when I tried it that the `stderr` output of my script is still swallowed. Also, when the script exits with non-zero error status, the error is ignored (with or without this workaround).

**Comment by mykeels** [comment]
That the method is overridable should be enough to implement error handling, right?

One of its parameters is the actual error string, if I understand correctly.

**Comment by rrthomas** [comment]
You're right, this should work, but via the `exit` event connected to `_onScriptComplete`, which does nothing in case of error.

Thanks for prompting me to look at the code, I think I can supply the missing PR now.

**Comment by drewloomer** [comment]
Published version `0.3.1`. Thanks, all!
